web/src/pages/checkout/index.js

```diff
@@ -0,0 +1,276 @@
+import dayjs from 'dayjs'
+import _ from 'underscore'
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { setTransaction as setStoreTransaction, makePruchase } from '../../store/modules/shop/actions'
+import './styles.css';
+import Header from '../../components/header';
+import Product from '../../components/product/list';
+
+
+
+
+function Checkout() {
+
+  const dispatch = useDispatch();
+
+  const { cart, transactionFee, defaultRecipient } = useSelector((state) => state.shop);
+
+  const total = cart.reduce((total, product) => {
+    return total + product.preco;
+  }, 0)
+
+  const [transaction, setTransaction] = useState({
+    amount: 0,
+    card_number: '',
+    card_cvv: '',
+    card_expiration_date: '',
+    card_holder_name: '',
+    shipping: {
+      name: 'Petfood LTDA',
+      fee: 1000,
+      delivery_date: dayjs().add(7, 'days').format('YYYY-MM-DD'),
+      expedited: true,
+      address: {
+        country: 'br',
+        state: '',
+        city: '',
+        neighborhood: '',
+        street: '',
+        street_number: '',
+        zipcode: '',
+      },
+    },
+    items: [],
+    split_rules: [],
+  });
+
+
+  const setShippingValue = (key, value) => {
+    setTransaction({
+      ...transaction,
+      shipping: {
+        ...transaction.shipping,
+        address: {
+          ...transaction.shipping.address,
+          [key]: value,
+        },
+      },
+    });
+  };
+
+  const makePurchase = () => {
+    dispatch(setStoreTransaction(transaction));
+    setTimeout(() => {
+      dispatch(makePruchase())
+    }, 100)
+  }
+
+  const getSplitRules = () => {
+    const productsByPetshop = _.groupBy(
+      cart,
+      (product) => product.petshop_id.recipient_id
+    );
+
+    let result = [];
+
+    Object.keys(productsByPetshop).map((petshop) => {
+      const products = productsByPetshop[petshop];
+      const totalValuePerPetshop = products
+        .reduce((total, product) => {
+          return total + product.preco;
+        }, 0)
+        .toFixed(2);
+
+      const totalFee = (totalValuePerPetshop * transactionFee).toFixed(2);
+
+      result.push({
+        recipient_id: products[0].petshop_id.recipient_id,
+        percentage: Math.floor(
+          ((totalValuePerPetshop - totalFee) / total) * 100
```

Comment:
Would be good to change the [magic numbers](https://flaviocopes.com/magic-numbers/) to constants